source/common/http/functional_stream_decoder_base.cc

```diff
@@ -334,5 +371,47 @@ void FunctionRetrieverMetadataAccessor::fetchClusterInfoIfOurs() {
   }
 }
 
+// register the factory as it is a must if you use the base filter
+
+class FunctionBaseFilterFactory
+    : public Extensions::HttpFilters::Common::EmptyHttpFilterConfig {
```

Comment:
I'm not sure if I like the inheritance here, given that the derived config class ends up not being empty.

Reply:
i'm not sure what you mean - as it doesnt have a filter config so its empty in that sense. can you elaborate?